server/mux.go

```diff
@@ -208,26 +212,41 @@ func AuthAPI(opts MuxOpts, router chronograf.Router) (http.Handler, AuthRoutes)
 	for _, pf := range opts.ProviderFuncs {
 		pf(func(p oauth2.Provider, m oauth2.Mux) {
 			urlName := PathEscape(strings.ToLower(p.Name()))
-			loginPath := fmt.Sprintf("%s/oauth/%s/login", opts.Basepath, urlName)
-			logoutPath := fmt.Sprintf("%s/oauth/%s/logout", opts.Basepath, urlName)
-			callbackPath := fmt.Sprintf("%s/oauth/%s/callback", opts.Basepath, urlName)
+
+			loginPath := path.Join("/oauth", urlName, "login")
+			logoutPath := path.Join("/oauth", urlName, "logout")
+			callbackPath := path.Join("/oauth", urlName, "callback")
+
 			router.Handler("GET", loginPath, m.Login())
 			router.Handler("GET", logoutPath, m.Logout())
 			router.Handler("GET", callbackPath, m.Callback())
 			routes = append(routes, AuthRoute{
-				Name:     p.Name(),
-				Label:    strings.Title(p.Name()),
-				Login:    loginPath,
-				Logout:   logoutPath,
-				Callback: callbackPath,
+				Name:  p.Name(),
+				Label: strings.Title(p.Name()),
+				// AuthRoutes are content served to the page. When Basepath is set, it
+				// says that all content served to the page will be prefixed with the
+				// basepath. Since these routes are consumed by JS, it will need the
+				// basepath set to traverse a proxy correctly
+				Login:    path.Join(opts.Basepath, loginPath),
+				Logout:   path.Join(opts.Basepath, logoutPath),
+				Callback: path.Join(opts.Basepath, callbackPath),
 			})
 		})
 	}
 
+	rootPath := "/chronograf/v1"
+	logoutPath := "/oauth/logout"
+
+	if opts.PrefixRoutes {
+		rootPath = path.Join(opts.Basepath, rootPath)
+		logoutPath = path.Join(opts.Basepath, logoutPath)
+	}
+
 	tokenMiddleware := AuthorizedToken(opts.Auth, opts.Logger, router)
 	// Wrap the API with token validation middleware.
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if strings.HasPrefix(r.URL.Path, "/chronograf/v1/") || r.URL.Path == "/oauth/logout" {
+		cleanPath := path.Clean(r.URL.Path) // compare ignoring path garbage, trailing slashes, etc.
+		if (strings.HasPrefix(cleanPath, rootPath) && len(cleanPath) > len(rootPath)) || cleanPath == logoutPath {
```

Comment:
Since this is a routing decision, whether `basepath` applies is determined by `PrefixRoutes`. That switch is done up here: https://github.com/influxdata/chronograf/pull/1407/files/57a0a304234264d1c6412992a566f554c2f6a85e#diff-54baed360b3d60b20e84d4d6a1cec940R242 . If `PrefixRoutes` isn't set, we assume that a load balancer has already stripped off the prefix, and we should proceed as though `Basepath`s didn't exist.